pages/activity/index.js

```diff
@@ -0,0 +1,77 @@
+import Head from "next/head";
+import { useState, useEffect } from "react";
```

Comment:
You are not using useEffect somewhere in your code. Generally consider not to useEffect quite often since most of the time you won´t need it